index.html

```diff
@@ -9,30 +9,36 @@
     <!-- Bootstrap CSS -->
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
 
+    <script src="https://code.jquery.com/jquery-3.1.0.slim.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0/angular.js"></script>
     <script src="js/app.js"></script>
+    <script src="js/ng-infinite-scroll.js"></script>
+    <link rel="stylesheet" href="css/styles.css">
 
-
-    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
-    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
-    <!--[if lt IE 9]>
-      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.2/html5shiv.min.js"></script>
-      <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
-    <![endif]-->
   </head>
-  <body ng-app="GiphyApp">
+  <body ng-app="GiphyApp" ng-controller="MainCtrl">
     <div class="container">
       <h1 class="text-center">Welcome to Giphy Search</h1>
 
       <form class="form-inline">
         <div class="form-group">
           <label for="search">Search</label>
-          <input type="text" placeholder="Search for GIFs" id="search" class="form-control">
+          <input type="text" id="search"
+          class="form-control" ng-model="searchterm">
         </div>
-        <button type="submit" class="btn btn-primary">Search</button>
+        <button type="submit" class="btn btn-primary" ng-click="search()">Search</button>
       </form>
 
-      <div id="results"></div>
+      <div ng-show="s404"><img src="img/404.gif" alt=""></div>
+
+      <div class="flex"
+           infinite-scroll="myPagingFunction()"
+           infinite-scroll-distance="1">
+        <div class="results" ng-repeat="result in results track by $index">
+          <img src="{{ result.images.fixed_height.url }}" alt="">
+        </div>
+      </div>
+
```

Comment:
Div element to host the infinite scrolling:

    <div class="row" id="results" infinite-scroll="myPagingFunction()" infinite-scroll-distance="3">
        <div class="well col-md-6" ng-repeat="giphy in giphys track by giphy.id">
            <img ng-src="{{giphy.images.fixed_height.url}}" alt="Description" />
            <a ng-href="{{giphy.images.fixed_height.url}}">link</a>
        </div>
    </div>